EIPS/eip-6900.md

```diff
@@ -0,0 +1,358 @@
+
+```solidity
+interface IModularAccount {
+   /**
+    * view function calls to the account
+    * @param implAddress the address of the implmentation contract
+    * @param data params to pass to the view function
+    */
+   function viewCall(address implAddress, bytes calldata data) external returns (bytes memory);
+}
+```
+
+Plugin modification interface. Modular Smart Contract Accounts MAY implement this interface to support updates to plugins.
+
+#### `IPluginUpdate.sol`
+
+```solidity
+interface IPluginUpdate {
+    enum PluginAction {
+        ADD,
+        REPLACE,
+        REMOVE
+    }
+
+    enum AssociatedFunctionType {
+        USER_OP_VALIDATOR,
+        RUNTIME_VALIDATOR,
+        PRE_HOOK,
+        POST_HOOK
+    }
+
+    struct PluginUpdate {
+        address executionPluginAddress;
+        PluginAction action;
+        ExecutionUpdate[] executionUpdates;
+    }
+
+    struct GlobalPluginUpdate {
+        PluginAction action;
+        AssociatedFunction globalPlugin;
+    }
+
+    struct ExecutionUpdate {
+        bytes4 executionSelector;
+        AssociatedFunction[] associatedFunctions;
+    }
+
+    struct AssociatedFunction {
+        AssociatedFunctionType functionType;
+        address implAddress;
+        bytes4 implSelector;
+    }
+
+    event PluginsUpdated(PluginUpdate[] pluginUpdates, address init, bytes callData);
+
+    event GlobalPluginsUpdated(AssociatedFunction[] globalPluginUpdates, address init, bytes callData);
+
+    /**
+        * @notice Add/replace/remove any number of plugins and optionally execute a function
+        * @param pluginUpdates Contains the plugin addresses and function selectors.
+        *        executionPluginAddress specifies the plugin containing the execution functions defined within
+        *        pluginAction denotes what operation to perform
+        *        executionUpdates denote which execution function and associated function to perform the opeartion on.
+        * @param init The address of the contract or facet to execute calldata
+        * @param callData A function call, including function selector and arguments
+        */
+    function updatePlugins(PluginUpdate[] memory pluginUpdates, address init, bytes calldata callData) external;
+
+    /**
+     * @notice Add/replace/remove any number of global plugins and optionally execute a function
+     * @param globalPluginUpdates Contains the plugin addresses and function selectors.
+     * @param init The address of the contract or facet to execute calldata
+     * @param callData A function call, including function selector and arguments
+     */
+   function updateGlobalPlugins(GlobalPluginUpdate[] memory globalPluginUpdates, address init, bytes calldata callData)
+       external;
+}
+```
+
+Plugin inspection interface. Modular Smart Contract Accounts MAY implement this interface to support visibility in plugin configuration on-chain. Note this will require extra storage.
+
+#### `IPluginLoupe.sol`
```

Comment:
```suggestion
#### `IPluginLoupe.sol`

Plugin inspection interface. Modular Smart Contract Accounts MAY implement this interface to support visibility in plugin configuration on-chain. Note this will require extra storage.
```